pkg/controller/builder.go

```diff
@@ -254,17 +253,23 @@ func UpdateDeploymentProxyHook(pc clients.ProxyClientInterface) deploymentcontro
 	}
 }
 
-func setContainerArg(con *corev1.Container, arg, value string) error {
-	if slices.ContainsFunc(con.Args, func(oldArg string) bool {
-		return strings.HasPrefix(oldArg, arg)
-	}) {
-		return fmt.Errorf("unexpected argument %q in container %q while building manifests", arg, con.Name)
```

Comment:
Before, we found the argument to be an error.
I am trying to understand this scenario.
Could you please provide an example of the type of argument we are removing?

Reply:
The experimental manifests will already have the `--feature-gate` arguments (because the feature kustomize component's resources/patches add them), so we should now expect those arguments in the manifest. So, we simply delete them to replace them with the arguments provided by OCP (TechPreview).

This function is _only_ called for the `--feature-gate` argument.

The new new comments added to this function. Are they not sufficient?

Reply:
make sense I got now